### PR TITLE
[New Design] Passcode Screen

### PR DESCRIPTION
**Pull request by davcres**
## Related Issues
App: #3516 

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here]
_____

## Screenshots
| Old view check passcode | New view check passcode | Old view create passcode | New view create passcode |
|:-:|:-:|:-:|:-:|
| Screenshot_20220311-094445_ownCloud | MicrosoftTeams-image | Screenshot_20220311-094502_ownCloud | MicrosoftTeams-image (1) |

## QA

Test plan: https://github.com/owncloud/QA/blob/master/Mobile/Android/Release_2.21/3582-SettingsSecurity.md

### CODE REVIEW

File: changelog/unreleased/3569 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Improvements for the UI in the passcode screen
+
+Redesign of the passcode screen to have the numeric keyboard in the screen instead of using the Android one.
+Linear and Relatives layouts were deleted, now only ConstraintLayout is used.
```
Comment (fesave) [inline changelog/unreleased/3569]: I think that this detailed information is not relevant to the user, think that there may be people who do not know anything about layout.

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/NumberKeyboard.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package com.owncloud.android.presentation.ui.security.passcode
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/NumberKeyboard.kt:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/NumberKeyboardListener.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.owncloud.android.presentation.ui.security.passcode
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/NumberKeyboardListener.kt:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/PassCodeActivity.kt [hunk 4]
```diff
@@ -230,6 +227,30 @@ class PassCodeActivity : AppCompatActivity(), IEnableBiometrics {
...
+        val numberOfPasscodeDigits = (passCodeViewModel.getPassCode()?.length ?: passCodeViewModel.getNumberOfPassCodeDigits())
+
+        if (passcode.length < numberOfPasscodeDigits && !binding.lockTime.isVisible) {
+            passcode.append(number.toString())
+            passCodeEditTexts[passcode.length - 1]?.text = Editable.Factory.getInstance().newEditable(number.toString())
+        }
+    }
+
+    override fun onBiometricButtonClicked() {
+        //TODO
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/PassCodeActivity.kt:240]: Was this functionality included in this task or will it be developed later?

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/SquareFrameLayout.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package com.owncloud.android.presentation.ui.security.passcode
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/passcode/SquareFrameLayout.kt:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/res/layout/passcodelock.xml [hunk 6]
```diff
@@ -63,26 +68,55 @@
...
+    <!--
+    <Button
+        android:id="@+id/btnCancel"
+        android:theme="@style/Button.Primary"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@android:string/cancel"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"/>
+        -->
```
Comment (fesave) [inline owncloudApp/src/main/res/layout/passcodelock.xml:121]: Try to keep the code as clean as possible, if the button is commented for any reason, I encourage you to add a comment about it. If it is not used, remove it.

File: owncloudApp/src/main/res/layout/numberkeyboard_layout.xml [hunk 7]
```diff
@@ -0,0 +1,265 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (fesave) [inline owncloudApp/src/main/res/layout/numberkeyboard_layout.xml:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/res/values-hdpi/dims.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (fesave) [inline owncloudApp/src/main/res/values-hdpi/dims.xml:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/res/values-xhdpi/dims.xml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (fesave) [inline owncloudApp/src/main/res/values-xhdpi/dims.xml:1]: You forgot to add the copyright to the file! Don't worry, ask us how to configure it and we will help you 😉

File: owncloudApp/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -260,6 +260,20 @@
     <string name="pass_code_wrong">Incorrect passcode</string>
     <string name="pass_code_error_set">An error occurred while setting the passcode lock</string>
     <string name="pass_code_error_remove">An error occurred while removing the passcode lock</string>
+    <string name="numberkeyboard_backspace_button_content_description">Backspace button</string>
+    <string name="numberkeyboard_biometric_button_content_description">Biometric button</string>
```
Comment (fesave) [inline owncloudApp/src/main/res/values/strings.xml:264]: Excellent 🚀

File: owncloudApp/src/main/res/drawable/numberkeyboard_key_bg_transparent.xml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape />
```
Comment (fesave) [inline owncloudApp/src/main/res/drawable/numberkeyboard_key_bg_transparent.xml:2]: Does this do anything? haha 😆

**Review by fesave — CHANGES_REQUESTED** [review]
Excellent work @davcres 🥇 ! I posted you a series of comments in this first review, any questions, ask us.

File: owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/PassCodeActivityTest.kt [hunk 12]
```diff
@@ -128,14 +124,28 @@ class PassCodeActivityTest {
         R.id.explanation.isDisplayed(false)
 
         // Check if required amount of input fields are actually displayed
-        with(R.id.passCodeTxtLayout) {
+        with(R.id.layout_code) {
```
Comment (fesave) [inline owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/PassCodeActivityTest.kt:127]: Just a note, when you change the Id of elements of the view already made, warn the rest of the team, sometimes certain tests can be broken 😉

**Review by fesave — COMMENTED** [review]
Just another note

File: owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/PassCodeActivityTest.kt (hunk 12 above)
Comment (davcres) [reply owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/PassCodeActivityTest.kt:127]: Okay, thank you

**Review by fesave — CHANGES_REQUESTED** [review]
Excellent work @davcres 🚀 , but... Could you check the XML files? You have several `@+id` to replace. Thanks 😄

**Comment by jesmrec** [comment]
Approved!

**Review by fesave — APPROVED** [review]
LGTM

**Pull request merged**